### ISSUE TITLE
Conversation doesn't work if the user doesn't have a username set

### ISSUE THREAD

**Issue by elie222** [benjick/meteor-telegram-bot#22]
Conversation doesn't work if the user doesn't have a username set.

Username is then `undefined` which breaks the example in the documentation, and even if you give some username, then `parsePollResult` will still break.

**Comment by fckngold** [comment]
Any progress?

**Comment by shrmnk** [comment]
Hey, I'll get started on this tomorrow. Sorry, was busy with some tests the past few weeks

**Comment by frranck** [comment]
@shrmnk Hi, I had no username and the example in the documentation didn't work, except the "TelegramBot.setCatchAllText(true, function(username, message)" 
I added an username in Telegram but the command examples still don't work for me... Any clue?

**Comment by shrmnk** [comment]
If you're free, could you check if this issue still persists on v1.3.0? I have no resources to set up a test environment at the moment so I can't work on them yet

**Comment by frranck** [comment]
@shrmnk thanks it's now working for me in v1.3.0 
 Didn't test without a username though...

**Comment by elie222** [comment]
Yeah. It was clear from the code that it wouldn't work without a username.
It's got nothing to do with the meteor version.

It just wasn't tested for the no username case which caused a bug. Probably
not a difficult fix

**Comment by frranck** [comment]
@elie222 the meteor-telegram-bot version is 1.3.0...

**Comment by elie222** [comment]
:) sorry about that. Then ignore what I said

**Comment by fckngold** [comment]
In TelegramBot.parsePollResult change `const fromUsername = item.message.from.username;` to `const fromUsername = item.message.from.id;` and in TelegramBot.startConversation remove `typeof(username) === 'string' &&`
Сan anybody confirm this?

**Comment by McBrick** [comment]
hi, thanks for this package..
i’d like to use this package, but i dunno the right way to use local package for overcome
this username problem, can anyone tell me?
or
can anyone amend the code similar to **MeDBejoHok**’s suggestion?

thank you

**Comment by elie222** [comment]
Go to your meteor project. Create a packages folder. Go to that folder in
the command line and do git clone the url for this package.

I think that's about it (you also need to add this package to your meteor
project). You can check it all worked if you do meteor list in the root of
your project and see a + symbol next to the package version. That means
it's a local package.

But I think pull requests were made to fix this no?

**Comment by McBrick** [comment]
@elie222 thx for your reply
i should have state it clearly..
i know how to add local package, work properly on local machine,
but when i push the project to server, it doesn't work, push fail..

**Comment by McBrick** [comment]
hi, more specifically, i use heroku and i got this error:
error: unknown package in top-level dependencies: [package_name]
does anyone know how to push to server properly? thanks.
